src/main/java/com/overops/plugins/service/impl/ReportBuilder.java

```diff
@@ -186,290 +189,198 @@ private Version createVersion(EventResult e) {
 		}
-		
+
 		//critical error gate
 		boolean critical = false;
-		if (qualityGateReport.getCriticalErrors() != null  && qualityGateReport.getCriticalErrors().size() > 0) {
+		if (criticalErrors.size() > 0) {
 			critical = true;
-			replaceSourceId(qualityGateReport.getCriticalErrors());
 		}
-		
+
 		//top errors
-		if (qualityGateReport.getTopErrors() != null  && qualityGateReport.getTopErrors().size() > 0) {
-			replaceSourceId(qualityGateReport.getTopErrors());
+		if (topErrors.size() > 0) {
 		}
-		
+
 		boolean checkCritical = false;
-		if (input.criticalExceptionTypes != null && input.criticalExceptionTypes.size() > 0) {
+		if (input.regressionInput.criticalExceptionTypes != null && input.regressionInput.criticalExceptionTypes.size() > 0) {
 			checkCritical = true;
 		}
-		
+
 		boolean unstable = (hasRegressions)
 				|| (maxVolumeExceeded)
 				|| (maxUniqueErrorsExceeded)
-				|| (newErrors)
+				|| (newError)
 				|| (resurfaced)
 				|| (critical);
-		
-		return new QualityReport(input, rateRegression, regressions,
-				qualityGateReport.getCriticalErrors(), qualityGateReport.getTopErrors(), qualityGateReport.getNewErrors(), 
-				qualityGateReport.getResurfacedErrors(), qualityGateReport.getTotalErrorCount(),
-				qualityGateReport.getUniqueErrorCount(), unstable, newEvents, resurfacedEvents, checkCritical, checkMaxEventGate,
-				checkUniqueEventGate, runRegressions, maxEventVolume, maxUniqueErrors, markedUnstable);
+		return new ReportBuilder.QualityReport(input.regressionInput, null, regressionErrors,
+				criticalErrors, topErrors, newErrors,
+				resurfacedErrors, totalErrorCount,
+				uniqueErrorCount, unstable, properties.isNewEvents(), properties.isResurfacedErrors(), checkCritical, checkMaxEventGate,
+				checkUniqueEventGate, runRegression, properties.getMaxErrorVolume(), properties.getMaxUniqueErrors(), properties.isMarkUnstable());
+	}
+
+	private static List<EventRow> filterEvents(List<SeriesRow> events, String regexFilter) {
+		if (regexFilter == null) {
+			return events.stream().map(e -> (EventRow) e).collect(Collectors.toList());
+		}
+
+		List<EventRow> returnEvents = new ArrayList<>();
+		for (SeriesRow event : events) {
+			if (evaluateEvent(event, getPattern(regexFilter))) {
+				EventRow e = (EventRow) event;
+				returnEvents.add(e);
+				// now increment counters
+				totalErrorCount += e.hits;
+				uniqueErrorCount++;
+			}
+		}
+		return returnEvents;
 	}
-	
-	//for each event, replace the source ID in the ARC link with the number 4 (which means Jenkins)
-	private static void replaceSourceId (List<OOReportEvent> events) {
-		for (OOReportEvent ooReportEvent : events) {
-			String arcLink = replaceSourceIdInArcLink(ooReportEvent.getARCLink());
-			ooReportEvent.setArcLink(arcLink);
+
+	private static boolean evaluateEvent(SeriesRow event, Pattern pattern) {
+		if (pattern == null) {
+			return true;
 		}
+
+		String json = new Gson().toJson(event);
+
+		return !pattern.matcher(json).find();
```

Comment:
can you please extract these to variables that state what that means, also makes debugging easier